### ISSUE TITLE
ubuntu make error

### ISSUE THREAD

**Issue by libaosuan** [colmap/colmap#619]
[100%] Linking CXX executable colmap
../libcolmap.a(bitmap.cc.o): In function `char const* boost::re_detail::re_is_set_member<char const*, char, boost::regex_traits<char, boost::cpp_regex_traits<char> >, unsigned int>(char const*, char const*, boost::re_detail::re_set_long<unsigned int> const*, boost::re_detail::regex_data<char, boost::regex_traits<char, boost::cpp_regex_traits<char> > > const&, bool)':
bitmap.cc:(.text._ZN5boost9re_detail16re_is_set_memberIPKccNS_12regex_traitsIcNS_16cpp_regex_traitsIcEEEEjEET_S8_S8_PKNS0_11re_set_longIT2_EERKNS0_10regex_dataIT0_T1_EEb[_ZN5boost9re_detail16re_is_set_memberIPKccNS_12regex_traitsIcNS_16cpp_regex_traitsIcEEEEjEET_S8_S8_PKNS0_11re_set_longIT2_EERKNS0_10regex_dataIT0_T1_EEb]+0x145): undefined reference to `boost::re_detail::cpp_regex_traits_implementation<char>::transform_primary[abi:cxx11](char const*, char const*) const'
bitmap.cc:(.text._ZN5boost9re_detail16re_is_set_memberIPKccNS_12regex_traitsIcNS_16cpp_regex_traitsIcEEEEjEET_S8_S8_PKNS0_11re_set_longIT2_EERKNS0_10regex_dataIT0_T1_EEb[_ZN5boost9re_detail16re_is_set_memberIPKccNS_12regex_traitsIcNS_16cpp_regex_traitsIcEEEEjEET_S8_S8_PKNS0_11re_set_longIT2_EERKNS0_10regex_dataIT0_T1_EEb]+0x4a1): undefined reference to `boost::re_detail::cpp_regex_traits_implementation<char>::transform[abi:cxx11](char const*, char const*) const'
collect2: error: ld returned 1 exit status
src/exe/CMakeFiles/colmap_exe.dir/build.make:150: recipe for target 'src/exe/colmap' failed
make[2]: *** [src/exe/colmap] Error 1
CMakeFiles/Makefile2:1184: recipe for target 'src/exe/CMakeFiles/colmap_exe.dir/all' failed
make[1]: *** [src/exe/CMakeFiles/colmap_exe.dir/all] Error 2
Makefile:127: recipe for target 'all' failed
make: *** [all] Error 2

ubuntu 16.04
cuda 9.0
gcc-6.5 g++-6.5

**Comment by maoyali** [comment]
The same problem，have you solved it？

**Comment by libaosuan** [comment]
> The same problem，have you solved it？

No.

**Comment by ogencoglu** [comment]
+1

**Comment by libaosuan** [comment]
how to solve it?

**Comment by silence401** [comment]
ubuntu18.04, the same error ...

**Comment by ahojnnes** [comment]
Which version of boost and CMake are you all using?

**Comment by hellocaicai96** [comment]
> how to solve it?

have you solved it yet?

**Comment by hellocaicai96** [comment]
> Which version of boost and CMake are you all using?

libboost1.58-dev:amd64 and cmake version 3.12.2

**Comment by libaosuan** [comment]
> Which version of boost and CMake are you all using?
libboost1.58 cmake 3.5.1

**Comment by libaosuan** [comment]
> Which version of boost and CMake are you all using?

libboost1.58 cmake 3.5.1

**Comment by ahojnnes** [comment]
Is there any chance you can upgrade to a more recent version of boost? I have both 16.04 and 18.04 running locally here with default packages and have no issues.

**Comment by GabbySuwichaya** [comment]
Hi! I am having the same problem. So, how this problem is solved ?

**Comment by miquelmassot** [comment]
Same problem here:

Ubuntu 16.04
gcc 6
Boost 1.58
CMake 3.5.1

```
[100%] Linking CXX executable colmap
cd /home/miquel/opt/colmap/build/src/exe && /usr/bin/cmake -E cmake_link_script CMakeFiles/colmap_exe.dir/link.txt --verbose=1
/usr/bin/g++-6    -Wno-maybe-uninitialized -fopenmp -std=c++11 -Wall -O3 -DNDEBUG   CMakeFiles/colmap_exe.dir/colmap.cc.o  -o colmap -rdynamic ../libcolmap.a ../../lib/FLANN/libflann.a ../../lib/Graclus/libgraclus.a ../../lib/LSD/liblsd.a ../../lib/PBA/libpba.a ../../lib/PoissonRecon/libpoisson_recon.a ../../lib/SQLite/libsqlite3.a ../../lib/SiftGPU/libsift_gpu.a ../../lib/VLFeat/libvlfeat.a ../libcolmap_cuda.a ../libcolmap.a -ldl -Wl,-Bstatic -lboost_filesystem -lboost_program_options -lboost_regex -lboost_system -lboost_program_options -lboost_filesystem -lboost_graph -lboost_regex -lboost_system -lboost_unit_test_framework -Wl,-Bdynamic -lglog -lfreeimage /usr/lib/libceres.so.1.12.0 -lGL -lGLU /usr/lib/x86_64-linux-gnu/libQt5OpenGL.so.5.5.1 /usr/lib/x86_64-linux-gnu/libQt5Widgets.so.5.5.1 /usr/lib/x86_64-linux-gnu/libCGAL.so.11.0.1 -lgmp -lpthread -lGLEW /usr/local/cuda-9.1/lib64/libcudart_static.a -ldl -lrt -Wl,-Bstatic -lboost_filesystem -lboost_program_options -lboost_graph -lboost_unit_test_framework -Wl,-Bdynamic -lfreeimage -lGL -lGLU /usr/lib/x86_64-linux-gnu/libCGAL.so.11.0.1 -lgmp -lpthread -lglog -lgflags -lpthread /usr/lib/x86_64-linux-gnu/libQt5Gui.so.5.5.1 /usr/lib/x86_64-linux-gnu/libQt5Core.so.5.5.1 
../libcolmap.a(bitmap.cc.o): In function `char const* boost::re_detail::re_is_set_member<char const*, char, boost::regex_traits<char, boost::cpp_regex_traits<char> >, unsigned int>(char const*, char const*, boost::re_detail::re_set_long<unsigned int> const*, boost::re_detail::regex_data<char, boost::regex_traits<char, boost::cpp_regex_traits<char> > > const&, bool)':
bitmap.cc:(.text._ZN5boost9re_detail16re_is_set_memberIPKccNS_12regex_traitsIcNS_16cpp_regex_traitsIcEEEEjEET_S8_S8_PKNS0_11re_set_longIT2_EERKNS0_10regex_dataIT0_T1_EEb[_ZN5boost9re_detail16re_is_set_memberIPKccNS_12regex_traitsIcNS_16cpp_regex_traitsIcEEEEjEET_S8_S8_PKNS0_11re_set_longIT2_EERKNS0_10regex_dataIT0_T1_EEb]+0x145): undefined reference to `boost::re_detail::cpp_regex_traits_implementation<char>::transform_primary[abi:cxx11](char const*, char const*) const'
bitmap.cc:(.text._ZN5boost9re_detail16re_is_set_memberIPKccNS_12regex_traitsIcNS_16cpp_regex_traitsIcEEEEjEET_S8_S8_PKNS0_11re_set_longIT2_EERKNS0_10regex_dataIT0_T1_EEb[_ZN5boost9re_detail16re_is_set_memberIPKccNS_12regex_traitsIcNS_16cpp_regex_traitsIcEEEEjEET_S8_S8_PKNS0_11re_set_longIT2_EERKNS0_10regex_dataIT0_T1_EEb]+0x4a1): undefined reference to `boost::re_detail::cpp_regex_traits_implementation<char>::transform[abi:cxx11](char const*, char const*) const'
collect2: error: ld returned 1 exit status
src/exe/CMakeFiles/colmap_exe.dir/build.make:140: recipe for target 'src/exe/colmap' failed
make[2]: *** [src/exe/colmap] Error 1
make[2]: Leaving directory '/home/miquel/opt/colmap/build'
CMakeFiles/Makefile2:1184: recipe for target 'src/exe/CMakeFiles/colmap_exe.dir/all' failed
make[1]: *** [src/exe/CMakeFiles/colmap_exe.dir/all] Error 2
make[1]: Leaving directory '/home/miquel/opt/colmap/build'
Makefile:127: recipe for target 'all' failed
make: *** [all] Error 2
```

**Comment by GabbySuwichaya** [comment]
miquelmassot, I didn't solve the problem. 

However, I tried reinstall my gcc, CUDA, and NVIDIA, and it works.

My CMake  and boost are at the same version as yours (Boost 1.58, CMake 3.5.1). 
For the rest, COLMAP's dependencies were set as follows: 
gcc/g++ version is 5.4.0 
CUDA 9.0
NVIDIA 384   

I think the problem that I had was that my previously installed CUDA and NVIDIA does not support one another (the NVIDIA and CUDA's versions were 418 and 9.0). And (probably) for this reason, after reinstalling gcc/g++, CUDA, and NVIDIA according to the list that I provided above, I got it to work.
